forum/views/comments.py

```diff
@@ -10,32 +10,87 @@
 from rest_framework.serializers import ValidationError
 from rest_framework.views import APIView
 
-from forum.models import Comment
+from forum.models import Comment, CommentThread
+from forum.models.model_utils import validate_object
 from forum.serializers.comment import UserCommentSerializer
 from forum.utils import str_to_bool
 
 
+def validate_comments_request_data(data: dict[str, Any]) -> None:
+    """
+    Validates the request data if it exists or not.
+
+    Parameters:
+        data: request data to validate.
+    Response:
+        raise exception if some data does not exists.
+    """
+    fields_to_validate = ["body", "course_id", "user_id"]
+    for field in fields_to_validate:
+        if field not in data or not data[field]:
+            raise ValueError(f"{field} is missing.")
+
+
+def create_comment(
+    data: dict[str, Any],
+    depth: int,
+    thread_id: Optional[str] = None,
+    parent_id: Optional[str] = None,
+) -> Any:
+    """handle comment creation and returns a comment"""
+    new_comment_id = Comment().insert(
+        body=data["body"],
+        course_id=data["course_id"],
+        anonymous=data.get("anonymous", False),
+        anonymous_to_peers=data.get("anonymous_to_peers", False),
+        author_id=data["user_id"],
+        comment_thread_id=thread_id,
+        parent_id=parent_id,
+        depth=depth,
+    )
+    return Comment().get(new_comment_id)
+
+
+def prepare_comment_api_response(
+    comment: dict[str, Any],
+    exclude_fields: Optional[list[str]] = None,
+) -> dict[str, Any]:
+    """
+    Return serialized validated data.
+
+    Parameters:
+        comment: The comment details that needs to be serialized.
+        exclude_fields: Any fields that need to be excluded from response.
+
+    Response:
+        serialized validated data of the comment.
+    """
+    comment_data = {
+        **comment,
+        "id": str(comment.get("_id")),
+        "user_id": comment.get("author_id"),
```

Comment:
I think we don't need this because we already set the source to the author_id.

Reply:
Same with the username

Reply:
the serializer raises the exception as we are using is_valid.